package.json

```diff
@@ -28,20 +34,24 @@
     "sinon": "^3.3.0",
     "temp": "^0.8.3",
     "tslint": "^5.12.0",
-    "tslint-language-service": "^0.9.9",
     "typedoc": "^0.15.0-0",
     "typedoc-plugin-external-module-map": "^1.0.0",
     "typescript": "~3.5.3",
     "uuid": "^3.2.1"
   },
   "scripts": {
+    "preinstall": "node-gyp install",
     "postinstall": "node scripts/post-install.js",
-    "prepare": "node scripts/skip-prepare || ( yarn prepare:travis && yarn rebuild:clean && yarn build:clean && yarn prepare:hoisting && yarn download:plugins)",
+    "prepare": "yarn prepare:travis && yarn prepare:build && yarn prepare:hoisting && yarn download:plugins",
     "prepare:travis": "node scripts/prepare-travis",
+    "prepare:build": "yarn build && run lint && run build \"@theia/example-*\" --stream --parallel",
     "prepare:hoisting": "theia check:hoisted -s",
-    "preinstall": "node-gyp install",
-    "build": "run build",
-    "build:clean": "run prepare",
+    "clean": "yarn lint:clean && node scripts/run-reverse-topo.js yarn clean",
+    "build": "tsc -b compile.tsconfig.json --verbose",
```

Comment:
Could we move compile.tsconfig.json under config? Generally any config which does not has to be in the root has to be moved to config.

Reply:
Unfortunately, I don't think we can. Something to do with `rootDir` and `outDir`.

Reply:
Unless you mean only the root one.

Reply:
Yes, only keeping root clean from many config files